src/execution/join_hashtable.cpp

```diff
@@ -41,16 +42,21 @@ JoinHashTable::JoinHashTable(BufferManager &buffer_manager, vector<JoinCondition
 	for (idx_t i = 0; i < build_types.size(); i++) {
 		build_size += GetTypeIdSize(build_types[i].InternalType());
 	}
-	tuple_size = condition_size + build_size;
+	validity_size = ValidityData::EntryCount(conditions.size() + build_types.size()) * sizeof(validity_t);
```

Comment:
Minimum 8-byte requirement per tuple feels a bit brutal here, especially since this will usually fit into 1 or 2 bytes. Perhaps we can byte-align the validity size instead?

A clean way of doing this might be to have a TemplatedValidityMask and TemplatedValidityData sit underneath the current ValidityMask/ValidityData, which is by default instantiated with `validity_t` (`uint64_t`), but can also be instantiated with a `uint8_t`.

Reply:
Yeah that was bugging me too. It seemed a rather large change to the whole code base, but if I'm going to fix the "static" method issue, I can do this too.

Reply:
One of the reasons I left it was alignment problems, but I think the correct fix there is to align after the data columns. All these changes make me think that my `RowChunk` management class is what is needed here. With that I could make the change in one place...

Reply:
The rest of the code-base should be able to stay the same, no? It would just mean the ValidityMask would become an alias for `TemplatedValidityMask<validity_t>`, i.e. you would have somewhere in the validity mask header `using ValidityMask = TemplatedValidityMask<validity_t>`.

Reply:
> One of the reasons I left it was alignment problems, but I think the correct fix there is to align after the data columns. All these changes make me think that my `RowChunk` management class is what is needed here. With that I could make the change in one place...

Alignment problems are fair, but the same alignment problems would happen if we had a 1-byte column (i.e. grouping on a `TINYINT` or so). This is why we use the Load and Store interface all over the place. Alignment problems should be caught by the undefined behavior sanitizer as well.

Reply:
I was jus thinning about the lack of alignment between the data and the aggregate states. `Load` and `Store` make alignment moot for the data values, but the stuff after them may not be so happy.

Incidentally, I was a bit perturbed by the way the initialised aggregate states are copied from a static buffer. It assumes that the states are POD, which might not be the case!

Reply:
> I was jus thinking about the lack of alignment between the data and the aggregate states. Load and Store make alignment moot for the data values, but the stuff after them may not be so happy.

Indeed care needs to be taken with aligning the payloads. I believe the payloads are 8-byte aligned later on though, but perhaps I am mistaken.  

> Incidentally, I was a bit perturbed by the way the initialised aggregate states are copied from a static buffer. It assumes that the states are POD, which might not be the case!
 
Yes, we could change this so the `init` is called once per aggregate entry (preferably in a vectorized manner, so we don't get a function call per entry, but rather one function call for all new entries).

Reply:
It looks like there are some `ValidityMask` methods (`Slice`, `Combine`, `Resize`) that are complex and would need to be pulled into the header. Is that really worth doing for this PR?

Reply:
Maybe the right thing to do is to use the template as a base class for `ValidityMask`. The base class would just be used for externally managed storage?

Reply:
Yes, that sounds good.

Reply:
Well it was more tangled than that, but I have managed to smack it around.